### PR TITLE
feat(argo-cd): Support annotations in argocd-configs secrets

### PR DESCRIPTION
**Pull request by yu-croco**
This PR resolves https://github.com/argoproj/argo-helm/issues/1280 🙋 

---

Signed-off-by: yu-croco <yuki.kita22@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/main/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -218,7 +218,9 @@ NAME: my-release
 | configs.knownHosts.data.ssh_known_hosts | string | See [values.yaml] | Known Hosts |
 | configs.knownHostsAnnotations | object | `{}` | Known Hosts configmap annotations |
 | configs.repositories | object | `{}` | Repositories list to be used by applications |
+| configs.repositoryCredentialSecretAnnotations | object | `{}` | Repository credential secret annotations |
 | configs.repositoryCredentials | object | `{}` | *DEPRECATED:* Instead, use `configs.credentialTemplates` and/or `configs.repositories` |
```
Comment (mkilchhofer) [inline charts/argo-cd/README.md:222]: `repositoryCredentialSecretAnnotations` follows the deprecated way to configure credentials.
Better use `configs.credentialTemplatesAnnotations` or something similar

File: charts/argo-cd/templates/argocd-configs/repository-secret.yaml [hunk 2]
```diff
@@ -7,6 +7,10 @@ metadata:
   labels:
     argocd.argoproj.io/secret-type: repository
     {{- include "argo-cd.labels" (dict "context" $) | nindent 4 }}
+  {{- with $.Values.configs.repositorySecretAnnotations }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-configs/repository-secret.yaml:10]: configs.repositoriesAnnotations

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

**Comment by yu-croco** [comment]
@mkilchhofer 
Thank you for reviewing! I fixed them on https://github.com/argoproj/argo-helm/pull/1314/commits/8ef71ae664a6124bc5eef388352b825ab3583323 🙋

**Review by mkilchhofer — APPROVED** [review]
Lgtm. Thanks for your contribution, @yu-croco ! Maybe you nominate yourself as a maintainer? :-) -> https://github.com/argoproj/argo-helm/issues/612

**Comment by yu-croco** [comment]
> Maybe you nominate yourself as a maintainer? :-) -> https://github.com/argoproj/argo-helm/issues/612

Thank you so much!
Yes, I am nominating myself 🙋

**Comment by yu-croco** [comment]
It looks this PR merged into other project.... 🙄 
I will fix this on this repository.

**Pull request merged**